cryptography/src/lib.rs

```diff
@@ -140,16 +140,19 @@ pub trait Digest: Array + Copy {
 }
 
 /// An object that can be uniquely represented as a [Digest].
-pub trait Digestible<D: Digest>: Clone + Sized + Send + Sync + 'static {
+pub trait Digestible: Clone + Sized + Send + Sync + 'static {
+    type Digest: Digest;
     /// Returns a unique representation of the object as a [Digest].
     ///
     /// If many objects with [Digest]s are related (map to some higher-level
     /// group [Digest]), you should also implement [Committable].
-    fn digest(&self) -> D;
+    fn digest(&self) -> Self::Digest;
 }
 
-/// An object that shares a (commitment) [Digest] with other, related values.
-pub trait Committable<D: Digest>: Clone + Sized + Send + Sync + 'static {
+/// An object that can produce a commitment of itself.
+pub trait Committable: Clone + Sized + Send + Sync + 'static {
```

Comment:
We could add `Committable: Digestible` to make:

```rs
pub struct Engine<E: Clock + Spawner + Metrics, P: Array, M: Committable + Digestible + Codec> {
```

into

```rs
pub struct Engine<E: Clock + Spawner + Metrics, P: Array, M: Committable + Codec> {
```

but I think it'd be OK to leave as is, too.

Reply:
As discussed offline, I think it makes sense to keep these separate 👍